File: Data Structures & Algorithms/house-robber/submission-5.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        vector<int> max_prices(n+1,0);
        max_prices[0] = 0;
        max_prices[1] = nums[0];
        int i = 2;
        while(i <= n){
            max_prices[i] = max(max_prices[i-1],nums[i-1]+max_prices[i-2]);
            i++;
        }
        return max_prices[n];
    }
};

File: Data Structures & Algorithms/hand-of-straights/submission-5.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        nums = {}
        curmin = float('inf')
        curmax = 0
    
        for i in hand:
            if i in nums:
                nums[i] += 1
            else:
                nums[i] = 1
            curmin = min(curmin,nums[i])
            curmax = max(curmax,nums[i])
        avail = set(nums.keys())
        
        while(avail):
            currmin = min(avail)
            i = 0
            while(i < groupSize and currmin+i in avail):
                nums[currmin+i] -= 1
                if nums[currmin+i] == 0:
                    avail.remove(currmin+i)
                i += 1
            if i <= groupSize -1:
                return False
        return True
                